src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3178,6 +3371,66 @@ GetAggregateType(Aggref *aggregateExpression)
 		}
 	}
 
+	/*
+	 * All functions from github.com/tvondra/tdigest start with the "tdigest" prefix.
+	 * Since it requires lookups of function names in a schema we would like to only
+	 * perform these checks if there is some chance it will actually result in a positive
+	 * hit.
+	 */
+	if (StartsWith(aggregateProcName, "tdigest"))
+	{
+		if (aggFunctionId == TDigestExtensionAggTDigest1())
```

Comment:
Why check each OID rather than having something like `AggregateNames` to check `aggregateProcName` against?

Reply:
the name of this aggergate is `tdigest` which collides between
`tdigest(tdigest)` and `tdigest(value, compression)`. Both will have `tdigest` as their `aggregateProcName`.

Same goes for `tdigest_percentile` (4 overloads) and `tdigest_percentile_of` (4 overloads).

To make sure we can distinguish between them it is easiest to look them up via their Oid.